server/core/src/actors/v1_read.rs

```diff
@@ -402,6 +403,53 @@ impl QueryServerReadV1 {
             })
     }
 
+    #[instrument(level = "debug", skip_all)]
+    /// pull an image so we can present it to the user
+    pub async fn handle_oauth2_rs_image_get_image(
+        &self,
+        uat: Option<String>,
+        rs: Filter<FilterInvalid>,
+    ) -> Result<ImageValue, OperationError> {
+        let mut idms_prox_read = self.idms.proxy_read().await;
+        let ct = duration_from_epoch_now();
+
+        let ident = idms_prox_read
+                .validate_and_parse_token_to_ident(uat.as_deref(), ct)
+                .map_err(|e| {
+                    admin_error!(err = ?e, "Invalid identity in handle_oauth2_rs_image_get_image {:?}", uat);
+                    e
+                })?;
+
+        let attrs: BTreeSet<AttrString> = vec![Attribute::Image.into()].into_iter().collect();
+
+        let filter_orig = filter_all!(f_self())
+            .validate(idms_prox_read.qs_read.get_schema())
+            .map_err(OperationError::SchemaViolation)?;
+        let filter = rs
+            .clone()
+            .validate(idms_prox_read.qs_read.get_schema())
+            .map_err(OperationError::SchemaViolation)?;
+
+        let search = SearchEvent {
```

Comment:
Avoid building this manually. I think you'll break access control here becaus filter_orig and filter have extremely precise and subtle meanings. 

There are searchEvent builders that can help you here, and you should use one of them.

The way this works is filter_orig is the "filter as given by the user". then filter is "the filter as we are going to actually execute it, with extra conditionals provided that MAY bypass the users access controls. 

This is that in filter we can wrap with class exclusions for tombstone/recycled, even if the account *can not* read "class". But then we evaluate access based on "filter_orig" which is what the user actually asked for. 

So in this case you bypass access controls because you do filter_orig (f_self()) which is allowed, but then filter as executed is completely different.

Reply:
And yes, the searchEvent builders methods do this all for you correctly, so you proably want `SearchEvent::from_internal_message(ident, rs, attrs, &idms_prox_read.qs_read)` here.

Reply:
Done!